### DISCUSSION TITLE
DynamoDBAdapter: Region is missing

### DISCUSSION THREAD

**Discussion by dmeehan1968** [nextauthjs/next-auth#11459] [Help]
This error is thrown when the `session: { strategy: 'database' } }` is explicitly specified or omitted, and the DynamoDBAdapter is in use.

```
[auth][debug]: adapter_getSessionAndUser {
  "args": [
    "692a34a0-d33d-4ae9-9dcf-b8cacbad8f07"
  ]
}
[auth][error] AdapterError: Region is missing. Read more at https://errors.authjs.dev#adaptererror
```

If the region is specified in the client constructor, it then complains about missing credentials.

However, if the `session: { strategy: 'jwt' } }` is used, the adapter works fine as my Nextjs distribution is already authorised (I'm using https://sst.dev/ion to deploy via AWS.

The adapter shouldn't be checking whether credentials are provided to the client as the environment might provide appropriate config.  It certainly shouldn't be only validating credentials for one part of the flow (the session creation).

**Comment by lixin-dev** [comment]
I met the same error
I followed the chatgpt's instruction, it seems solved by using fromNodeProviderChain()

```
import { fromNodeProviderChain } from "@aws-sdk/credential-providers";

// ✅ use fromNodeProviderChain() 
const client = DynamoDBDocument.from(
  new DynamoDB({
    credentials: fromNodeProviderChain(), 
    region: process.env.AWS_REGION!,
  }),
  {
    marshallOptions: {
      convertEmptyValues: true,
      removeUndefinedValues: true,
      convertClassInstanceToMap: true,
    },
  }
);

export const { handlers, auth, signIn, signOut } = NextAuth({
  providers: [],
  session: {
        // forced to db
        strategy: "database",
    },
  adapter: DynamoDBAdapter(client),
});
```
